### ISSUE TITLE
Config validator rejects section headers with a trailing comment

### ISSUE THREAD

**Issue by ahgraber** [rvben/rumdl-vscode#177]
The extension's config validator recognizes a TOML section header only when the trimmed line ends at the closing bracket.
A trailing comment defeats the match, and the scanner then keeps the previous section's schema active, so every key that follows is reported as an unknown property of the wrong section.
The rumdl binary parses the same file without complaint, which makes the diagnostics look like a linter disagreement rather than an editor bug.

Seen with `rvben.rumdl` 0.0.291 (darwin-arm64) against rumdl 0.2.43.

## Reproduction

`.rumdl.toml`:

```toml
[global]
line_length = 80

[MD013] # Line length
reflow = true
```

Open the file in VS Code. `reflow` is flagged:

```text
Unknown property 'reflow' in [global] section
```

`rumdl check .` reports no config warning for the same file. 

## Workaround

The workaround is admittedly simple (and perhaps sufficient for this to be a #wontfix, which I would totally understand):

```toml
### Line length
[MD013]
reflow = true
```

**Comment by rvben** [comment]
@ahgraber thanks for the report, and for the analysis. It was exact.

Fixed in v0.0.307, now live on the Marketplace and Open VSX. Not a wontfix.

The header match required the line to end at the closing bracket, so `[MD013] # Line length` was not recognised as a header, `[global]` stayed active, and `reflow` was reported against the wrong section. Section headers now accept a trailing comment.

While in there I fixed the case that shares the cause. A bracketed line the scanner does not recognise as a rumdl section, an array of tables such as `[[tool.mypy.overrides]]` for instance, also used to leave the previous section active, so the keys under it were attributed to whatever came before. Those keys now go unvalidated rather than being flagged against the wrong schema.

Your workaround is no longer needed.
